src/Sentry/PlatformAbstractions/FrameworkInfo.NetFx.cs

```diff
@@ -0,0 +1,195 @@
+#if NETFX
+using System;
+using System.Collections.Generic;
+using Microsoft.Win32;
+
+namespace Sentry.PlatformAbstractions
+{
+    /// <summary>
+    /// Information about .NET Framework in the running machine
+    /// </summary>
+    public static partial class FrameworkInfo
+    {
+        /// <summary>
+        /// Get the latest Framework installation for the specified CLR
+        /// </summary>
+        /// <remarks>
+        /// Supports the current 3 CLR versions:
+        /// CLR 1 => .NET 1.0, 1.1
+        /// CLR 2 => .NET 2.0, 3.0, 3.5
+        /// CLR 4 => .NET 4.0, 4.5.x, 4.6.x, 4.7.x
+        /// </remarks>
+        /// <param name="clrVersion">The CLR version: 1, 2 or 4</param>
+        /// <returns>The framework installation or null if none is found.</returns>
+        public static FrameworkInstallation? GetLatest(int clrVersion)
+        {
+            // CLR versions
+            // https://docs.microsoft.com/en-us/dotnet/standard/clr
+            if (clrVersion != 1 && clrVersion != 2 && clrVersion != 4)
+            {
+                return null;
+            }
+
+            if (clrVersion == 4)
+            {
+                var release = Get45PlusLatestInstallationFromRegistry();
+                if (release != null)
+                {
+                    return new FrameworkInstallation
+                    {
+                        Version = GetNetFxVersionFromRelease(release.Value),
+                        Release = release
+                    };
+                }
+            }
+
+            FrameworkInstallation latest = null;
+            foreach (var installation in GetInstallations())
+            {
+                if (latest == null)
+                {
+                    latest = installation;
+                }
+
+                if (clrVersion == 2)
+                {
+                    // CLR 2 runs .NET 2 to 3.5
+                    if ((installation.Version.Major == 2 || installation.Version.Major == 3)
+                        && installation.Version >= latest.Version)
+                    {
+                        latest = installation;
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+                else if (clrVersion == 4)
+                {
+                    if (installation.Version.Major == 4
+                        && installation.Version >= latest.Version)
+                    {
+                        latest = installation;
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+            }
```

Comment:
Can this loop be replaced with `latest = installations.OrderBy(...).FirstOrDefault()` to avoid mutations and to make it simpler?